src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+                        alice["USD"](100),
+                        XRP(100),
+                        tfImmediateOrCancel),
+                    seq + 3),
+                batch::inner(pay(alice, bob, XRP(100)), seq + 4),
+                batch::inner(pay(alice, carol, XRP(100)), seq + 5),
+                batch::inner(pay(alice, eve, XRP(100)), seq + 6));
+
+            auto const txIDs = env.tx()->getBatchTransactionIDs();
+            TxID const parentBatchId = env.tx()->getTransactionID();
+            std::vector<TestBatchData> testCases = {
+                {"tecKILLED", to_string(txIDs[0])},
+                {"tecKILLED", to_string(txIDs[1])},
+                {"tecKILLED", to_string(txIDs[2])},
+                {"tesSUCCESS", to_string(txIDs[3])},
+            };
+            env.close();
+            validateBatch(env, parentBatchId, testCases);
+
+            BEAST_EXPECT(env.balance(alice) == preAlice - XRP(100) - batchFee);
+            BEAST_EXPECT(env.balance(bob) == preBob + XRP(100));
+            BEAST_EXPECT(env.balance(carol) == preCarol);
+        }
+    }
+
+    void
+    testInnerSubmitRPC(FeatureBitset features)
+    {
+        testcase("inner submit rpc");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+
+        env.fund(XRP(1000), alice, bob);
+        env.close();
+        // Invalid RPC Submission: TxnSignature
+        // - has `TxnSignature` field
+        // - has no `SigningPubKey` field
+        // - has no `Signers` field
+        // - has `tfInnerBatchTxn` flag
+        {
+            auto jv = pay(alice, bob, XRP(1));
+            jv[sfFlags.fieldName] = tfInnerBatchTxn;
+            Serializer s;
+            auto jt = env.jt(jv);
+            jt.stx->add(s);
+            auto const jrr = env.rpc("submit", strHex(s.slice()))[jss::result];
+            BEAST_EXPECT(
+                jrr[jss::status] == "error" &&
+                jrr[jss::error] == "invalidTransaction" &&
+                jrr[jss::error_exception] ==
+                    "fails local checks: Malformed: Invalid inner batch "
+                    "transaction.");
+
+            env.close();
+        }
+
+        // Invalid RPC Submission: SigningPubKey
+        // - has no `TxnSignature` field
+        // - has `SigningPubKey` field
+        // - has no `Signers` field
+        // - has `tfInnerBatchTxn` flag
+        {
+            std::string txBlob =
+                "1200002240000000240000000561D4838D7EA4C68000000000000000000000"
+                "0000005553440000000000A407AF5856CCF3C42619DAA925813FC955C72983"
+                "68400000000000000A73210388935426E0D08083314842EDFBB2D517BD4769"
+                "9F9A4527318A8E10468C97C0528114AE123A8556F3CF91154711376AFB0F89"
+                "4F832B3D8314F51DFC2A09D62CBBA1DFBDD4691DAC96AD98B90F";
```

Comment:
Can this transaction be submitted from JSON like the first one? Easier to read/audit that way.

Reply:
Ditto for the other blobs